src/com/facebook/buck/jvm/groovy/GroovycStep.java

```diff
@@ -91,14 +98,61 @@ public String getDescription(ExecutionContext context) {
   }
 
   private ImmutableList<String> createCommand() {
-    ImmutableList.Builder<String> command = ImmutableList.builder();
-    command.addAll(groovyc.getCommandPrefix(resolver))
+    final ImmutableList.Builder<String> command = ImmutableList.builder();
+
+    command.addAll(groovyc.getCommandPrefix(resolver));
+
+    String classpath = Joiner.on(":").join(transform(declaredClasspathEntries, toStringFunction()));
+    command
         .add("-cp")
-        .add(Joiner.on(":").join(transform(declaredClasspathEntries, toStringFunction())))
+        .add(classpath.isEmpty() ? "''" : classpath)
         .add("-d")
-        .add(outputDirectory.toString())
-        .addAll(extraArguments.or(ImmutableList.<String>of()))
-        .addAll(transform(sourceFilePaths, toStringFunction()));
+        .add(outputDirectory.toString());
+    addCrossCompilationOptions(command);
+
+    command.addAll(extraArguments.or(ImmutableList.<String>of()))
+           .addAll(transform(sourceFilePaths, toStringFunction()));
     return command.build();
   }
+
+  private void addCrossCompilationOptions(final ImmutableList.Builder<String> command) {
+    if (shouldCrossCompile()) {
+      command.add("-j");
+      javacOptions.appendOptionsTo(new OptionsConsumer() {
+        @Override
+        public void addOptionValue(String option, String value) {
+          // javac won't find things compiled with groovyc otherwise.
```

Comment:
otherwise?  It seems like some text might be missing here or it is misleading me how it is worded.